sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -117,14 +120,166 @@ private[hive] class HiveClientImpl(
     case hive.v3_1 => new Shim_v3_1()
   }
 
+  def getCurrentState(): SessionState = {
+    val ugi = UserGroupInformation.getCurrentUser
+    // TODO need to check if we should add more condition
+    if (ugi.getAuthenticationMethod() == UserGroupInformation.AuthenticationMethod.PROXY) {
+      getOrCreateSessionState(ugi.getShortUserName)
+    } else {
+      state
+    }
+  }
+
```

Comment:
If it's not proxy, use origin state and client. keep same action without this pr.
If proxy, one user use one `Hive` client